internals-js/src/utils.ts

```diff
@@ -479,3 +479,29 @@ export function setsEqual<T>(s1: Set<T> | null, s2: Set<T> | null): boolean {
   }
   return true;
 }
+
+// ordering is done based on order that elements were added
+export class TimeOrderedSet<V extends { name: string }> {
+  private _values: V[] = [];
+  private _valueMap: Map<string, V> = new Map();
```

Comment:
In ECMAScript/JS, `Map` and `Set` are specified to preserve insertion order when you iterate over them, so you might not need this wrapper class?

Reply:
Interesting. I didn't realize that. I'll change it.